dashboard/app/models/user.rb

```diff
@@ -2075,6 +2077,32 @@ def destroy
     end
   end
 
+  # Called before_destroy.
+  # Soft-deletes any projects and other channel-backed progress belonging to
+  # this user.  Unfeatures any featured projects belonging to this user.
+  private def soft_delete_channels
+    return unless user_storage_id
+
+    channel_ids = PEGASUS_DB[:storage_apps].
+      where(storage_id: user_storage_id).
+      map(:id)
+
+    # Unfeature any featured projects owned by the user
+    FeaturedProject.
+      where(storage_app_id: channel_ids).
+      update_all(unfeatured_at: Time.now)
+
+    # Soft-delete all of the user's channels
+    PEGASUS_DB[:storage_apps].
+      where(id: channel_ids).
+      update(state: 'deleted')
+  end
+
+  # Gets the user's user_storage_id from the pegasus database, if it's available.
+  def user_storage_id
+    @user_storage_id ||= PEGASUS_DB[:user_storage_ids].where(user_id: id).first&.[](:id)
```

Comment:
this is already implemented here: https://github.com/code-dot-org/code-dot-org/blob/05908d48ec4b36064a60d5a704086ac27257bc31/shared/middleware/helpers/storage_id.rb#L92-L95

Reply:
that's terrible :-( I'd be fine with you proceeding the way you already have working. then, one path to fix this atrocity is to remove `storage_id_for_user_id` from `test_helper`, see what breaks, and then stub `storage_id_for_user_id` only for the tests which broke.